sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,193 @@
+import numpy as np
+from ...utils import check_matplotlib_support
+
+
+def _check_boundary_response_method(estimator, response_method):
+    """Return prediction method from the response_method for decision boundary
+
+    Parameters
+    ----------
+    estimator: object
+        Estimator to check
+
+    response_method: {'auto', 'predict_proba', 'decision_function', 'predict'}
+        Specifies whether to use :term:`predict_proba` or
+        :term:`decision_function` as the target response. If set to 'auto',
+        :term:`predict_proba` is tried first and if it does not exist
+        :term:`decision_function` is tried next.
+
+    Returns
+    -------
+    prediction_method: callable
+        prediction method of estimator
+    """
+
+    if response_method not in ("predict_proba", "decision_function",
+                               "auto", "predict"):
+        raise ValueError("response_method must be 'predict_proba', "
+                         "'decision_function', 'predict', or 'auto'")
+
+    error_msg = "response method {} is not defined in {}"
+    if response_method != "auto":
+        if not hasattr(estimator, response_method):
+            raise ValueError(error_msg.format(response_method,
+                                              estimator.__class__.__name__))
+        return getattr(estimator, response_method)
+    elif hasattr(estimator, 'predict_proba'):
```

Comment:
right now "classifier comparison" does decision_function, then predict_proba. For plotting I think that actually makes sense since a softmax will always compress down the boundary.
I assume you got this from metrics, where having probabilities might be preferred.